### ISSUE TITLE
Rule "collapsable if statements" too aggressive?

### ISSUE THREAD

**Issue by maplesteve** [oclint/oclint#10]
Hi!

Consider the following code:

``` c
if (OUTER_TEST_1) {
    if (INNER_TEST) {
        doSomething();
    } else {
        doSomethingDifferent();
    }
} else if (OUTER_TEST_2) {
   (...)
}
```

The oclint rule complains / suggests, that OUTER_TEST_1 and INNER_TEST should be combined into just one line.
Maybe I just don't see the optimization here, but I think, that the statements can't be combined since doSomethingDifferent shall only be reached, if INNER_TEST is false but OUTER_TEST_1 is true.

Is my code 'wrong' or should the rule 'look' a little bit further in the code?
